Automattic/jetpack
# renovate: Run self-hosted via GitHub Actions

#### Changes proposed in this Pull Request:

The switch to using `.pnpmfile.cjs` for overrides in #22965 turned out
to make renovate unhappy, as they [intentionally don't use that file][1].

The solution is to run renovate "self-hosted", which we can do easily
enough via GitHub Actions. That also gives us a few other benefits:

* No need for the "renovate helper" workflow anymore, as we can
  configure postUpgradeTasks with a script to do the needful.
* No need to manually update the list of monorepo projects in
  renovate.json anymore, because we can use a .js config file for the
  self-hosted setup.
* That also lets us use comments in the config file.

The down side is that we're going to have to set up an account or app
for renovate to run as with a token that has the "workflow" scope. The
existing matticbot token doesn't have it.

[1]: https://github.com/renovatebot/renovate/discussions/11250

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check that the dry run done on push to the branch seems like it will be doing the right things.
  * :heavy_check_mark: https://github.com/Automattic/jetpack/actions/runs/1959383070
* Fork the repo, merge this, and run it in non-dry mode.
  * :heavy_check_mark: It created anomiex/jetpack#2, anomiex/jetpack#3, and anomiex/jetpack#4.
* Fix the stuff noted TODO, merge it, then poke at renovate stuff and see if it still works.
